src/main/java/randoop/generation/Bloodhound.java

```diff
@@ -0,0 +1,384 @@
+        // prevent
+        // ourselves from immediately re-updating branch coverage information should it be the case
+        // that the next test sequence that is generated is not a regression test and thus
+        // totalSuccessfulInvocations is not recomputed causing shouldUpdateBranchCoverage to be
+        // true
+        // again.
+        if (shouldUpdateBranchCoverage) {
+          totalSuccessfulInvocations = 1;
+        }
+        break;
+      default:
+        throw new BugInRandoopException(
+            "Unhandled value for bloodhound_update_mode: "
+                + GenInputsAbstract.bloodhound_update_mode);
+    }
+
+    if (shouldUpdateBranchCoverage) {
+      if (GenInputsAbstract.bloodhound_logging) {
+        System.out.println("Updating branch coverage information.");
+      }
+
+      methodSelectionCounts.clear();
+      coverageTracker.updateBranchCoverageMap();
+      updateWeightsForAllOperations();
+      logMethodWeights();
+    }
+  }
+
+  /** For debugging, print all method weights to standard output. */
+  private void logMethodWeights() {
+    if (GenInputsAbstract.bloodhound_logging) {
+      System.out.println("Method name: method weight");
+      for (TypedOperation typedOperation : methodWeights.keySet()) {
+        System.out.println(typedOperation.getName() + ": " + methodWeights.get(typedOperation));
+      }
+      System.out.println("--------------------------");
+    }
+  }
+
+  /**
+   * Computes and updates weights in {@code methodWeights} map for all methods under test.
+   * Recomputes the {@code totalWeightOfMethodsUnderTest} to avoid problems with round-off error.
+   */
+  private void updateWeightsForAllOperations() {
+    double totalWeight = 0;
+    for (TypedOperation operation : operationSimpleList) {
+      totalWeight += updateWeight(operation);
+    }
+    totalWeightOfMethodsUnderTest = totalWeight;
+  }
+
+  /**
+   * Recompute weight for a method under test. A method under test is assigned a weight based on a
+   * weighted combination of
+   *
+   * <ul>
+   *   <li>the number of branches uncovered and
+   *   <li>the ratio between the number of times this method has been recently selected and the
+   *       maximum number of times any method under test has been successfully invoked.
+   * </ul>
+   *
+   * The weighting scheme is based on Bloodhound in the Guided Random Testing (GRT) paper.
+   *
+   * @param operation method to compute weight for
+   * @return the updated weight for the given operation
+   */
+  private double updateWeight(TypedOperation operation) {
+    // Remove type arguments, because Jacoco does not include type arguments when naming a method.
+    String methodName = operation.getName().replaceAll("<.*>\\.", ".");
+
+    // Corresponds to uncovRatio(m) in the GRT paper.
+    Double uncovRatio = coverageTracker.getBranchCoverageForMethod(methodName);
+
+    if (uncovRatio == null) {
+      // Default to zero for methods with no coverage information.
```

Comment:
"Default to zero" is inconsistent with `uncovRatio = 0.5;` on line 303.  Which one is desired?